packages/@aws-cdk/toolkit-lib/lib/toolkit/toolkit.ts

```diff
@@ -865,6 +910,47 @@ export class Toolkit extends CloudAssemblySourceBuilder {
     return ret;
   }
 
+  /**
+   * Refactor Action. Moves resources from one location (stack + logical ID) to another.
+   */
+  public async refactor(cx: ICloudAssemblySource, options: RefactorOptions = {}): Promise<void> {
+    const ioHelper = asIoHelper(this.ioHost, 'refactor');
+    const assembly = await assemblyFromSource(ioHelper, cx);
+    return this._refactor(assembly, ioHelper, options);
+  }
+
+  private async _refactor(assembly: StackAssembly, ioHelper: IoHelper, options: RefactorOptions = {}): Promise<void> {
+    if (!options.dryRun) {
+      throw new ToolkitError('Refactor is not available yet. Too see the proposed changes, use the --dry-run flag.');
+    }
+
+    const strategy = options.stacks?.strategy ?? StackSelectionStrategy.ALL_STACKS;
+    if (strategy !== StackSelectionStrategy.ALL_STACKS) {
+      await ioHelper.notify(IO.CDK_TOOLKIT_W8010.msg(
```

Comment:
Shouldn't this just be an error?

If you want to convert user input into a warning and proceeding anyway, then do the conversion in the CLI?